### PR TITLE
chore: add metadata to decode endpoint

### PR DESCRIPTION
**Pull request by alexruzenhack**
### Acceptance Criteria
- extende the decode endpoint `/wallet/decode` with transaction, input, output and balance metadatas

### Security Checklist
- [x] Make sure you do not include new dependencies in the project unless strictly necessary and do not include dev-dependencies as production ones. More dependencies increase the possibility of one of them being hijacked and affecting us.

Closes: https://github.com/HathorNetwork/hathor-wallet-headless/issues/304

### CODE REVIEW

File: src/controllers/wallet/wallet.controller.js [hunk 1]
```diff
@@ -347,11 +355,55 @@ async function decodeTx(req, res) {
...
+
+    // True if all the inputs are signed, false otherwise
+    data.completeSignatures = data.inputs.length > 0
+      ? data.inputs.every(input => input.signed) // true until find a false statement
+      : false; // empty data.inputs
+
+    // Get balance
+    const balance = {};
+    { // Default HTR balance.
+      const balanceObj = await req.wallet.getBalance(hathorLibConstants.HATHOR_TOKEN_CONFIG.uid);
```
Comment (r4mmer) [inline src/controllers/wallet/wallet.controller.js:372]: This is not the balance of the transaction, it is the balance of the entire wallet.
I think you should use [getTxBalance](https://github.com/HathorNetwork/hathor-wallet-lib/blob/master/src/utils/transaction.ts#L209) instead

File: src/controllers/wallet/wallet.controller.js [hunk 2]
```diff
@@ -347,11 +355,55 @@ async function decodeTx(req, res) {
...
+        authorities: {
+          melt: { available: 0, locked: 0 },
+          mint: { available: 0, locked: 0 },
+        }
+      });
+    }
+
+    // Other token balance if any
+    for (const token of data.tokens) {
+      const balanceObj = await req.wallet.getBalance(token);
```
Comment (r4mmer) [inline src/controllers/wallet/wallet.controller.js:387]: Same here

**Review by r4mmer — CHANGES_REQUESTED** [review]

File: src/controllers/wallet/wallet.controller.js (hunk 1 above)
Comment (alexruzenhack) [reply src/controllers/wallet/wallet.controller.js:372]: Maybe I have misunderstood this. I thought I would return wallet metadata as well, not only transaction metadata. That's why I return the wallet balance.

**Review by r4mmer — APPROVED** [review]

File: src/helpers/tx.helper.js [hunk 3]
```diff
@@ -201,8 +201,42 @@ async function prepareTxFunds(wallet, outputs, inputs, defaultToken = HATHOR_TOK
   };
 }
 
+/**
+ * Best effort to get a transaction. First look up at the storage to get a trasaction
```
Comment (tuliomir) [inline src/helpers/tx.helper.js:205]:
```suggestion
 * Best effort to get a transaction. First look up at the storage to get a transaction
```

File: src/helpers/tx.helper.js [hunk 4]
```diff
@@ -201,8 +201,42 @@ async function prepareTxFunds(wallet, outputs, inputs, defaultToken = HATHOR_TOK
...
+ * If the transaction is not found in the fullnode, return null.
+ *
+ * @param {HathorWallet} wallet
+ * @param {string} id Hash of the transaction to get data from
+ * @return {Promise<DecodedTx|FullNodeTx|null>} Data from the transaction to get.
+ *    Can be null if both the wallet and fullnode does not contain the tx.
+ *
+ * @see DecodedTx at {@link https://github.com/HathorNetwork/hathor-wallet-lib/blob/bc94221cece2bd6d7b64d971ef30b7d593f07e42/src/new/wallet.js#L1058}
+ * @see FullNodeTx at {@link https://github.com/HathorNetwork/hathor-wallet-lib/blob/bc94221cece2bd6d7b64d971ef30b7d593f07e42/src/wallet/types.ts#L500}
+ */
```
Comment (tuliomir) [inline src/helpers/tx.helper.js:216]: :star2: Nice reference style!

**Review by tuliomir — APPROVED** [review]

**Pull request merged**